pkg/controllers/nodeclass/status/ami.go

```diff
@@ -44,6 +44,10 @@ func (a *AMI) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (reconc
 		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeAMIsReady, "AMINotFound", "AMISelector did not match any AMIs")
 		return reconcile.Result{}, nil
 	}
+	// Find if any discovered AMI is deprecated
+	containsDeprecatedAMIs := lo.SomeBy(amis, func(ami amifamily.AMI) bool {
```

Comment:
nit: Can we shift this `lo.SomeBy` call to the condition where it's actually used? You may even be able to parse through the `nodeClass.Status.AMIs` rather than the returned amis -- though either is probably fine